Flutter/pdf-viewer/undo-redo.md

```diff
@@ -0,0 +1,74 @@
+---
+layout: post
+title: Undo and Redo actions on the annotations in Flutter PDF Viewer widget | Syncfusion
+description: Learn here all about Undo and Redo actions on the annotations in Syncfusion Flutter PDF Viewer (SfPdfViewer) widget and more.
+platform: Flutter
+control: SfPdfViewer
+documentation: ug
+---
+
+# Undo and Redo actions on the annotations in Flutter PDF Viewer widget (Syncfusion)
+
+If you performed any undesired actions when on adding, removing or editing annotations, you can undo and redo the action to restore the previous state. This section will go through how to perform the undo and redo the changes made on the annotations.
+
+For desktop platforms such as Windows, macOS and desktop web, you can also use the following shortcut keys to perform the actions.
+
+<table>
+<tr>
+<th>Action & Shortcut keys</th>
+<th>Windows</th>
+<th>macOS</th>
+</tr>
+<tr>
+<th>Undo</th>
+<td><code>Ctrl</code> + <code>z</code></td>
+<td><code>Command</code> + <code>z</code></td>
+</tr>
+<tr>
+<th>Redo</th>
+<td><code>Ctrl</code> + <code>y</code></td>
+<td><code>Command</code> + <code>y</code></td>
+</tr>
+</table>
+
+You can perform undo and redo operations in the `SfPdfViewer` by assigning the `UndoHisotyController` instance to the `undoController` property of the `SfPdfViewer`. The `UndoHistoryController` class contains the `undo` and `redo` methods to perform the undo and redo operations respectively. The `canUndo` and `canRedo` properties are used to check whether the undo and redo operations can be performed respectively. The followwing code example illustrates how to perform undo and redo operations programmatically in the `SfPdfViewer` with the help of `UndoHistoryController` class.
```

Comment:
You can perform the undo and redo operations in the `SfPdfViewer` by assigning the `UndoHisotyController` instance to the `undoController` property of the `SfPdfViewer`. The `UndoHistoryController` class contains the `undo` and `redo` methods to perform the undo and redo operations, respectively. The `canUndo` and `canRedo` properties are used to check whether the undo and redo operations can be performed, respectively. The following code example illustrates how to perform the undo and redo operations programmatically in the `SfPdfViewer` with the help of the `UndoHistoryController` class.